src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+
+    private List<Entry<K, V>>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this.buckets = new ArrayList[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        ensureCapacity();
+        int index = Math.abs(hash(key)) % buckets.length;
+        List<Entry<K, V>> bucket = buckets[index];
+        if (bucket == null) {
+            bucket = new ArrayList<>();
+            buckets[index] = bucket;
+        }
+        for (Entry<K, V> entry : bucket) {
+            if (key == null && entry.key == null
+                    || key != null && key.equals(entry.key)) {
+                entry.value = value;
+                return;
+            }
+        }
+        bucket.add(new Entry<>(key, value));
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = Math.abs(hash(key)) % buckets.length;
+        List<Entry<K, V>> bucket = buckets[index];
+        if (bucket != null) {
+            for (Entry<K, V> entry : bucket) {
+                if ((key == null && entry.key == null)
+                        || (key != null && key.equals(entry.key))) {
```

Comment:
redundant parentheses 
```suggestion
                if (key == entry.key == null
                        || key != null && key.equals(entry.key)) {
```

Reply:
У виразі (key == null && entry.key == null) перевіряється, що і key, і entry.key рівні null. Це означає, що обидві змінні повинні бути null, щоб умова була істинною.

Проте в виразі key == entry.key == null спочатку порівнюються key і entry.key, а потім результат цього порівняння порівнюється з null. Такий вираз не буде коректним з точки зору семантики, і більшість мов програмування видасть помилку або результат не буде таким, як очікується.

Reply:
Сорі, недовиправив
Мало бути як в Objects.equals
```suggestion
                if (key == entry.key
                        || key != null && key.equals(entry.key)) {
```